tests/models/clip/test_modeling_clip.py

```diff
@@ -167,8 +184,174 @@ def prepare_config_and_inputs_for_common(self):
         return config, inputs_dict
 
 
+class CLIPModelTesterMixin(ModelTesterMixin):
```

Comment:
Why are we having to re-implement this for CLIP? Maybe a comment?

Reply:
Because we need to provide different inputs and compare different logits for different models, I added a comment in https://github.com/huggingface/transformers/pull/31940/commits/801f003562cf5adc6b38500062ec1c746b290b7d